android/sdl_android/src/main/java/com/smartdevicelink/util/SdlAppInfo.java

```diff
@@ -122,11 +139,66 @@ public String toString() {
         builder.append("\nLast updated: ");
         builder.append(this.lastUpdateTime);
 
+        builder.append("\nVehicle make list: ");
+        builder.append(this.vehicleMakesList.toString());
+
         builder.append("\n-------- Sdl App Info End------");
 
         return builder.toString();
     }
 
+    public static List<VehicleType> deserializeVehicleMake(XmlResourceParser parser) {
+        List<VehicleType> vehicleMakesList = new ArrayList<VehicleType>();
+
+        try {
+            int eventType = parser.getEventType();
+            VehicleType vehicleMake = null;
+            String data = null;
+            while (eventType != XmlPullParser.END_DOCUMENT) {
+                String tagname = parser.getName();
+                switch (eventType) {
+                    case XmlPullParser.START_TAG:
+                        if (tagname.equalsIgnoreCase("vehicle-type")) {
+                            // create a new instance of employee
+                            vehicleMake = new VehicleType();
+                        }
+                        break;
+
+                    case XmlPullParser.TEXT:
+                        data = parser.getText();
+                        break;
+
+                    case XmlPullParser.END_TAG:
+                        if (vehicleMake == null) {
+                            break;
+                        }
+                        if (tagname.equalsIgnoreCase("vehicle-type")) {
+                            // add employee object to list
+                            vehicleMakesList.add(vehicleMake);
+                        } else if (tagname.equalsIgnoreCase("make")) {
+                            vehicleMake.setMake(data == null ? "" : data);
+                        } else if (tagname.equalsIgnoreCase("model")) {
```

Comment:
@kostyaBoss For lines 180-186 are these fields `model`, `modelYear`, `trim` optional?  Do you need to check if these are `null` as well?